### PR TITLE
feat: Add new keyword to check max length based on space type (proposal body)

### PR DESCRIPTION
**Pull request by ChaituVR**
This will add a new keyword 
`maxLengthWithSpaceType` to allow higher body length limit to turbo spaces

```JSON
"maxLengthWithSpaceType": {
  "default": 10000,
  "turbo": 20000
}
```

so on sequencer or UI while calling `validateSchema` we can pass a new param `spaceType` which can be `default` by default or if we pass `turbo` max length will be `20000`

How to test:
- Run `yarn test` and Try passing bigger `body` to `proposal.json`
- and try adding `spaceType` to `test/schema.spec.ts`
```javascript 
const isValid = validateSchema(schema, example, {
  snapshotEnv: 'mainnet',
  spaceType: 'turbo'
});
```

### CODE REVIEW

**Comment by bonustrack** [comment]
We will need to allow more voting strategies too for Turbo spaces

**Comment by ChaituVR** [comment]
> We will need to allow more voting strategies too for Turbo spaces

Yes will create a new PR if we agree on the concept of adding new keyword like here 
follow status here https://discord.com/channels/955773041898573854/1187284357799481484/1187287152426504282

File: test/schema.spec.ts [hunk 1]
```diff
@@ -18,7 +20,44 @@ describe.each([
...
+      snapshotEnv: 'mainnet'
+    });
     expect(isValid).toBe(true);
   });
 });
+
+// Tests for turbo spaces
+
+describe.each([
+  { schemaType: 'space', schema: schemas.space, example: space },
```
Comment (wa0x6e) [inline test/schema.spec.ts:33]: Why do you test the space validation here, as the given `space` is not a Turbo space ?

**Review by wa0x6e — CHANGES_REQUESTED** [review]

File: test/schema.spec.ts [hunk 2]
```diff
@@ -18,7 +20,44 @@ describe.each([
...
+    expect(isValid).toBe(true);
+  });
+});
+
+describe.each([
+  // { schemaType: 'space', schema: schemas.space, example: space },
+  {
+    schemaType: 'proposal',
+    schema: schemas.proposal,
+    example: proposalTurbo,
```
Comment (wa0x6e) [inline test/schema.spec.ts:50]: This file name is confusing, as it's named turbo, but will not pass any turbo test

**Review by wa0x6e — CHANGES_REQUESTED** [review]

File: test/schema.spec.ts (hunk 1 above)
Comment (ChaituVR) [reply test/schema.spec.ts:33]: Yes we will have turbo on space too, changed it here https://github.com/snapshot-labs/snapshot.js/pull/951/files#diff-e755a2ecce42f09829d5c7dc1de8853d1d00ef56eaadc2709601c87b9be8ddfb So i thought better to add a placeholder

File: test/schema.spec.ts (hunk 2 above)
Comment (ChaituVR) [reply test/schema.spec.ts:50]: Sorry I don't get what you mean 🙈 

Here we test if turbo payload returns error.

**Review by wa0x6e — APPROVED** [review]
utAck

**Pull request merged**